osu.Game/Rulesets/Mods/ModTimeRamp.cs

```diff
@@ -56,9 +62,17 @@ public virtual void ApplyToBeatmap(Beatmap<T> beatmap)
         public virtual void Update(Playfield playfield)
         {
             var absRate = Math.Abs(FinalRateAdjustment);
-            var adjustment = MathHelper.Clamp(absRate * ((clock.CurrentTime - beginRampTime) / finalRateTime), 0, absRate);
+            applyAdjustment(MathHelper.Clamp(absRate * ((clock.CurrentTime - beginRampTime) / finalRateTime), 0, absRate));
+        }
+
+        private void applyAdjustment(double adjustment)
+        {
+            var localAdjust = 1 + Math.Sign(FinalRateAdjustment) * adjustment;
 
-            pitchAdjust.PitchAdjust = 1 + Math.Sign(FinalRateAdjustment) * adjustment;
+            if (clock is IHasPitchAdjust tempo)
+                tempo.PitchAdjust = baseAdjust * localAdjust;
+            else
```

Comment:
No consideration for `IHasTempoAdjust`?

Reply:
Yeah, because this mod is supposed to ramp pitch. There are no cases where we support tempo and not pitch. If you're not happy with this, should the interfaces be combined?

Reply:
Right ok, that's fine then